File: dev-docs/snippets/document-blob/prefetch.ts
import type { JsBaoClient } from "js-bao-wss-client";

export async function prefetch(
  client: JsBaoClient,
  documentId: string,
  blobIds: string[],
) {
  // #region example
  const blobs = client.document(documentId).blobs();
  // Pre-download blobs into the local cache for offline access.
  await blobs.prefetch(blobIds, { concurrency: 4 });
  // #endregion example
}

File: dev-docs/snippets/document-blob/prefetch.swift
import JsBaoClient

func prefetch(
  client: JsBaoClient,
  documentId: String,
  blobIds: [String]
) async throws {
  // #region example
  let blobs = client.documents.blobs(documentId: documentId)
  // Pre-download blobs into the local cache for offline access. Best-effort:
  // per-blob failures are swallowed. `concurrency` defaults to 2.
  await blobs.prefetch(blobIds: blobIds, concurrency: 4)
  // #endregion example
}
